### PR TITLE
ShipmentView: Implement Page/View with Tests

### PR DESCRIPTION
**Pull request by vahidbazzaz**
The PR implements page/view for the view shipment with test cases 4.5

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[2.0 ShipmentView: implement page/view with tests](https://trello.com/c/InYuNcFb)

File: react/src/views/Transfers/ShipmentView/ShipmentView.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        <Button
+          mt={10}
+          type="button"
+          borderRadius="0"
+          w="full"
+          variant="solid"
+          backgroundColor="red.300"
+          color="white"
+        >
+          Reject
```
Comment (pylipp) [inline react/src/views/Transfers/ShipmentView/ShipmentView.tsx:156]: I suppose clicking this button will call `cancelShipment`. Will it be part of another PR?

File: react/src/views/Transfers/ShipmentView/ShipmentView.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  Heading,
+  VStack,
+  Flex,
+  Spacer,
+  ButtonGroup,
+  Button,
+  Alert,
+  AlertIcon,
+} from "@chakra-ui/react";
+// import { useErrorHandling } from "hooks/error-handling";
```
Comment (pylipp) [inline react/src/views/Transfers/ShipmentView/ShipmentView.tsx:15]: Comments might be cleaned up, also lines 87 onwards :sparkles:

File: react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+              <Text fontStyle="italic" p={2}>
+                “{shipment.transferAgreement?.comment}”
+              </Text>
+              <Spacer />
+            </Center>
+          )}
+        </Box>
+        <Box p={4}>
+          <Center alignContent="stretch">
+            <Text fontWeight="bold">TOTAL:</Text> {shipment.details.length} BOX
```
Comment (pylipp) [inline react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx:105]: @HaGuesto and I just agreed that FE must filter for all details with `deletedOn === null`

Comment (vahidbazzaz) [reply react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx:105]: @pylipp I am pushing latest changes now, please check that version. :)

File: react/src/views/Transfers/ShipmentView/components/ShipmentTabs.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { TabList, TabPanels, Tabs, TabPanel, Tab, Center } from "@chakra-ui/react";
+import _ from "lodash";
+import { ShipmentDetail } from "types/generated/graphql";
+import ShipmentContent, { IShipmentContent } from "./ShipmentContent";
+
+export interface IShipmentTabsProps {
+  shipmentDetail: ShipmentDetail[];
+}
+function ShipmentTabs({ shipmentDetail }: IShipmentTabsProps) {
+  const boxGroupedByProductGender = _.values(
```
Comment (pylipp) [inline react/src/views/Transfers/ShipmentView/components/ShipmentTabs.tsx:10]: :exploding_head: for all this mapping! good job

File: react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx (hunk 3 above)
Comment (pylipp) [reply react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx:106]: :+1: also in the design it's `boxes` (lowercase and plural), and there should be space between `TOTAL:` and the count

File: react/src/views/Transfers/ShipmentView/ShipmentView.tsx (hunk 1 above)
Comment (vahidbazzaz) [reply react/src/views/Transfers/ShipmentView/ShipmentView.tsx:152]: @pylipp need to check this with @aerinsol and @HaGuesto

File: react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx (hunk 3 above)
Comment (vahidbazzaz) [reply react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx:106]: @pylipp I have not change it as @aerinsol working on the ui for that part

**Review by pylipp — APPROVED** [review]
Functional and code review done :heavy_check_mark: 
I like the structure of the components. Test cases are implemented acc. to test plan.
I'm fine with merging this until a next UI iteration comes around.
